src/stores/ViewportStore.ts

```diff
@@ -49,48 +62,64 @@ export default class ViewportStore extends Store<ViewportStoreState> {
-                // transitionInterpolator: new FlyToInterpolator({speed: 1.2}),
-                // transitionDuration: 'auto'
-                // todo: for some reason fitbounds can return zoom < 0 and the map is not visible when we load the
-                //       map without path
-                zoom: Math.max(0, zoom),
-            }
+        } else if (action instanceof InfoReceived) {
+            return calculateLatLngFromBbox(state, action.result.bbox)
+        } else if (action instanceof RouteRequestSuccess) {
+            // this assumes that always the first path is selected as result. One could use the
+            // state of the routeStore as well but then we would have to make sure that the route
+            // store digests this action first, which our Dispatcher can't at the moment.
+            return calculateLatLngFromBbox(state, action.result.paths[0].bbox!)
+        } else if (action instanceof SetSelectedPath) {
+            return calculateLatLngFromBbox(state, action.path.bbox!)
+        } else if (action instanceof PathDetailsRangeSelected) {
+            // take bbox from actio or from selected path
+            const bbox = action.bbox ? action.bbox : this.routeStore.state.selectedPath.bbox
+            // if neither has a bbox just fall through to unchanged state
+            if (bbox) return calculateLatLngFromBbox(state, bbox)
         }
         return state
     }
 }
 
+function calculateLatLngFromBbox(state: ViewportStoreState, bbox: Bbox): ViewportStoreState {
+    const bounds: [[number, number], [number, number]] = [
+        [Math.max(-179, bbox[0]), Math.max(-89, bbox[1])],
+        [Math.min(179, bbox[2]), Math.min(89, bbox[3])],
+    ]
+    const { longitude, latitude, zoom } = new WebMercatorViewport({
+        width: state.width,
+        height: state.height,
+    }).fitBounds(bounds, {
+        padding: getPadding(state.width, state.height),
+    })
+    return {
+        ...state,
+        longitude,
+        latitude,
+        transitionDuration: 500,
+        transitionInterpolator: new FlyToInterpolator(),
+        // there is also this option:
+        // transitionEasing: (t : any) => t,
+        // todo: we could also use speed/auto instead
+        // transitionInterpolator: new FlyToInterpolator({speed: 1.2}),
+        // transitionDuration: 'auto'
+        // todo: for some reason fitbounds can return zoom < 0 and the map is not visible when we load the
+        //       map without path
+        zoom: Math.max(0, zoom),
+    }
+}
+
 function getPadding(width: number, height: number) {
     const padding = mediaQuery.matches
         ? {
-              top: 250,
-              bottom: 150,
+              top: 200,
+              bottom: 32,
               right: 16,
               left: 16,
           }
         : {
               top: 100,
-              bottom: 100,
+              bottom: 300,
               right: 100,
               left: 500,
           }
```

Comment:
Nice! This fixed the problem that sometimes the markers were hidden by the path detail diagram, see #53